pkg/services/user/userimpl/user.go

```diff
@@ -594,3 +598,25 @@ func (s *Service) supportBundleCollector() supportbundles.Collector {
 		Fn:                collectorFn,
 	}
 }
+
+func hashUserIdentifier(identifier string, secret string) string {
+	key := []byte(secret)
+	h := hmac.New(sha256.New, key)
+	h.Write([]byte(identifier))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func buildUserAnalyticsSettings(signedInUser user.SignedInUser) user.AnalyticsSettings {
+	var settings user.AnalyticsSettings
+
+	if signedInUser.ExternalAuthID != "" {
+		settings.Identifier = signedInUser.ExternalAuthID
+	} else {
+		settings.Identifier = signedInUser.Email
+	}
+
+	if os.Getenv("GF_ANALYTICS_SECRET") != "" {
```

Comment:
2 questions:
- is this the best way to get a secret from the environment? 
- what should this be called? is `GF_ANALYTICS_SECRET` ok?

Reply:
No, we don't read the environment directly but rather read from `pkg/setting/setting.go`

In your case, you should add something like `intercom_secret` to

https://github.com/grafana/grafana/blob/27635e6f7bbbe04cd9fdfdca9feacf97267145e5/pkg/setting/setting.go#L395-L403

and

https://github.com/grafana/grafana/blob/27635e6f7bbbe04cd9fdfdca9feacf97267145e5/pkg/setting/setting.go#L1029-L1036

And to the `defaults.ini` and then copy over to `sample.ini` and prefix the non-comment lines with `;` (if that sounds a bit unclear, compare the two files visually and it should be clear) somewhere here

https://github.com/grafana/grafana/blob/27635e6f7bbbe04cd9fdfdca9feacf97267145e5/conf/defaults.ini#L220

`GF_SECTION_CONFIGURATION_OPTION` will automatically be created from reading the `defaults.ini`.